### PR TITLE
ETFE-2202 CAM-G04 guarantor-vat

### PR DESCRIPTION
**Pull request by grahampaulcook**

### CODE REVIEW

**Comment by grahampaulcook** [comment]
test this please

File: app/forms/sections/guarantor/GuarantorVatFormProvider.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import play.api.data.Form
+
+import javax.inject.Inject
+
+class GuarantorVatFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[String] =
+    Form(
+      "value" -> text("guarantorVat.error.required")
+        .verifying(maxLength(12, "guarantorVat.error.length"))
```
Comment (LewisMitchell) [inline app/forms/sections/guarantor/GuarantorVatFormProvider.scala:30]: This is probably me being naive. However, in the regex it says the max length is 14 chars?

File: conf/messages.en [hunk 2]
```diff
@@ -544,6 +544,19 @@ guarantorName.error.length = Business name must be 182 characters or less
 guarantorName.error.invalidCharacter = Business name must not contain < and > and : and ;
 guarantorName.change.hidden = business name
 
+guarantorVat.3.title = What is the goods owner''s VAT registration number?
+guarantorVat.3.heading = What is the goods owner''s VAT registration number?
+guarantorVat.4.title = What is the transporter''s VAT registration number?
+guarantorVat.4.heading = What is the transporter''s VAT registration number?
```
Comment (LewisMitchell) [inline conf/messages.en:550]: Can we use back ticks `’` instead?

File: test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        val request = FakeRequest(GET, controllers.sections.guarantor.routes.GuarantorVatController.onNonGbVAT(testErn, testLrn).url)
+
+        val result = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual onwardRoute.url
+      }
+    }
+
+    "must redirect to the guarantor arranger controller for a GET if no guarantor arranger value is found" in {
```
Comment (LewisMitchell) [inline test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala:142]: Is it valuable to have a test when a POST is performed and there is no guarantor arranger value in the user answers?

File: test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+            .withFormUrlEncodedBody(("value", "answer"))
+
+        val result = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual onwardRoute.url
+      }
+    }
+
+    "must redirect to the next page when the NONGBVAT link" in {
```
Comment (LewisMitchell) [inline test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala:117]:
```suggestion
    "must redirect to the next page when the NONGBVAT link is clicked" in {
```

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Comment by LewisMitchell** [comment]
Sorry, one more thing. There is no autocomplete attribute on the `input` element which is a AA WCAG 2.2 failure (according to ARC toolkit):
image

File: conf/messages.en (hunk 2 above)
Comment (grahampaulcook) [reply conf/messages.en]: Updated on https://github.com/hmrc/emcs-tfe-create-movement-frontend/pull/84/commits/f8220f4c3c2f8f72aa291200232000d80e7ffd49

File: app/forms/sections/guarantor/GuarantorVatFormProvider.scala (hunk 1 above)
Comment (grahampaulcook) [reply app/forms/sections/guarantor/GuarantorVatFormProvider.scala:30]: hmmm, https://github.com/hmrc/emcs-tfe-create-movement-frontend/blob/main/app/forms/sections/firstTransporter/FirstTransporterVatFormProvider.scala has 12 although https://github.com/hmrc/emcs-tfe-create-movement-frontend/blob/main/app/forms/sections/transportArranger/TransportArrangerVatFormProvider.scala has 14. I believe the schema for this is 14 (just checking that)

File: test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala (hunk 3 above)
Comment (grahampaulcook) [reply test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala:142]: For the edge case where a user has multiple tabs open and goes back to a tab and submits the page (but on another tab they had removed the guarantor earlier answers)

Comment (grahampaulcook) [reply test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala:143]: Added new test on https://github.com/hmrc/emcs-tfe-create-movement-frontend/pull/84/commits/200043256a2d0ed855a4ad8eec404e1d65df6c19

File: app/forms/sections/guarantor/GuarantorVatFormProvider.scala (hunk 1 above)
Comment (grahampaulcook) [reply app/forms/sections/guarantor/GuarantorVatFormProvider.scala:30]: Schema says `  <xs:simpleType name="VatNumberType">
    <xs:annotation>
      <xs:documentation>
        <doc:description value="VAT Number" />
      </xs:documentation>
    </xs:annotation>
    <xs:restriction base="xs:token">
      <xs:maxLength value="14" />
      <xs:pattern value=".{1,14}" />
    </xs:restriction>
  </xs:simpleType>`

Comment (LewisMitchell) [reply app/forms/sections/guarantor/GuarantorVatFormProvider.scala:30]: (or whatever country code)

File: test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala (hunk 3 above)
Comment (grahampaulcook) [reply test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala:143]: Updated in https://github.com/hmrc/emcs-tfe-create-movement-frontend/pull/84/commits/6d0fd9e7bfd6a21b0e37acc5510ecc31c5ed5b0b

File: app/forms/sections/guarantor/GuarantorVatFormProvider.scala (hunk 1 above)
Comment (grahampaulcook) [reply app/forms/sections/guarantor/GuarantorVatFormProvider.scala]: Updated in https://github.com/hmrc/emcs-tfe-create-movement-frontend/pull/84/commits/6d0fd9e7bfd6a21b0e37acc5510ecc31c5ed5b0b

Comment (LewisMitchell) [reply app/forms/sections/guarantor/GuarantorVatFormProvider.scala]: We can always change it back once the demo is done

**Comment by grahampaulcook** [comment]
> autocomplete
If thats needed then we are missing it from quite a few other places, the other 2 sections for VAT number do not have autocomplete enabled.

**Review by LewisMitchell — APPROVED** [review]
